### DISCUSSION TITLE
How to set up the upstream DNS server through the keyword of the regular expression instead of the domain name?

### DISCUSSION THREAD

**Discussion by hzqim** [AdguardTeam/AdGuardHome#4261] [Q&A] [answered]
On dnsmasq with a regex patch, I can define all the Google's domain name by the following syntax:
`server=/:(google|gstatic|ggpht|appspot|gmail|ytimg|youtu|gvt):/127.0.0.1#56`

And AdH does not match to the correct domain name with the following code :
`[/google/gstatic/ggpht/appspot/gmail/ytimg/youtu/gvt/]127.0.0.1#56`

Someone help me?
Thx！

**Comment by ainar-g** [comment, marked as answer]
Hello.  This is currently not supported, only domain names are.  See [this article][1] on our Wiki and the following feature request: #2432.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains

**Reply by hzqim** [reply]
Thanks for answer.
